### PR TITLE
Pr/fem abstraction layer implementation

### PR DESCRIPTION
**Pull request by aprochwicz**

### CODE REVIEW

File: src/fal/nrf_802154_fal.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#ifndef NRF_802154_FAL_H_
+#define NRF_802154_FAL_H_
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+extern int8_t nrf_802154_fal_tx_power_get(const uint8_t channel, const int8_t power);
```
[the remark this answers is not in the document]
Comment (aprochwicz) [reply src/fal/nrf_802154_fal.h]: `fal' is an abbreviation from `FEM Abstraction Layer`, so in this context `fal` is analogical to `raal`.

[the remark this answers is not in the document]
Comment (KaSroka) [reply src/fal/nrf_802154_fal.h]: In this file in line 32 you have `Radio Arbiter Abstraction Layer interface` instead of `FEM Abstraction Layer`

**Comment by aprochwicz** [comment]
Regression test

**Comment by aprochwicz** [comment]
Regression test

File: src/fal/nrf_802154_fal.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/* Copyright (c) 2018, Nordic Semiconductor ASA
```
Comment (hubertmis) [inline src/fal/nrf_802154_fal.c]: 2019

File: src/fal/nrf_802154_fal.c [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ */
+
+/**
+ * @file
+ *   This file contains implementation of the nRF 802.15.4 FEM abstraction layer.
+ *
+ */
+
+#include <assert.h>
```
Comment (hubertmis) [inline src/fal/nrf_802154_fal.c]: This include seems unnecessary here

**Comment by aprochwicz** [comment]
Regression test

**Comment by hubertmis** [comment]
Regression test

**Pull request merged**